applications/crossbar/test/cb_storage_test.erl

```diff
@@ -20,10 +64,113 @@ maybe_check_storage_settings_test_() ->
+                       ,<<"key">> => <<"my_AWS_ACCESS_KEY">>
+                       ,<<"secret">> => <<"my_AWS_SECRET_KEY">>
+                       }
+                 }
+           }
+     ,<<"id">> => ?ACCOUNT_ID_UNDER_TEST
+     ,<<"plan">> =>
+          #{<<"modb">> =>
+                #{<<"types">> =>
+                      #{<<"mailbox_message">> =>
+                            #{<<"attachments">> =>
+                                  #{<<"handler">> => UUID
+                                   }
+                             }
+                       }
+                 }
+           }
+     }.
+
 maybe_check(Context, HTTPVerb) ->
     cb_storage:maybe_check_storage_settings(Context, HTTPVerb).
+
+%% this is required for voice mail messages which their timestamp is missing from metadata and
+%% their private media is missing or both private media and mailbox doesn't have create/modified
+this_month_db_exists() ->
+    fun(Server, Db) ->
+            %% why meck can't call the original mfa properly?
+            %% meck_code_gen:get_current_call/1 returns undefined and it cause bad_match
+            %% in meck:passthrough/1
+            (kz_datamgr:db_classification(Db) =:= 'modb'
+             andalso is_db_under_test(Db, kazoo_modb:get_modb(kz_util:format_account_id(Db))))
+                orelse erlang:apply('kz_fixturedb_db_meck_original', db_exists, [Server, Db])
+    end.
+
+is_db_under_test(ThisMonth, ThisMonth) ->                                             true;
+is_db_under_test(<<"account%2Fac%2Fco%2Funt0000000000000000000000001-201710">>, _) -> true;
+is_db_under_test(_, _) ->                                                             false.
+
+-define(GET_LEGACY_CALL, {kz_datamgr, get_results, [?ACCOUNT_DB_UNDER_TEST, ?LEGACY_VIEW, [{limit, 2000}, descending]]}).
+
+%% Checking history calls  to kz_datamgr:get_results to see if any calls happens to
+%% get legacy messages, if yes return empty result to stop the process.
+check_kz_datamgr_history() ->
+    fun(Db, ?LEGACY_VIEW=View, Options) ->
+            History = meck:history(kz_datamgr),
+
+            %% meck:history returns:
+            %% * {CallerPid, MFA, Result}
+            %% * {CallerPid, MFA, ExceptionClass, ExceptionReason, StackTrace}
+            case lists:keyfind(?GET_LEGACY_CALL, 2, History) of
+                {_Pid, _MFA, {ok, ViewResult}} when length(ViewResult) == 10 ->
+                    %% ViewResult length is 10. This should be same as what is inside vmboxes+legacy_msg_by_timestamp.json
+                    %% in ?ACCOUNT_DB_UNDER_TEST FixtureDb directory.
+                    {ok, []};
+                _ ->
+                    meck:passthrough([Db, View, Options])
+            end;
+       (Db, View, Options) ->
+            meck:passthrough([Db, View, Options])
+    end.
+
+-spec open_cache_doc(kz_term:ne_binary(), kz_term:ne_binary()) ->
+                            {ok, kz_json:object()}.
+open_cache_doc(_DbName, _DocId) ->
```

Comment:
After mecking `kz_fixturedb_doc:open_doc` check the `DbName` to see it is the current month (`kazoo_modb:get_modb(kz_util:format_account_id(Db)))`) and if it is return success.

Since this is test is only interested in attachment of a document in current month db, by checking db for this month only, `open_doc` will work for other db as normal.